docs/specs/code/EXPORT_TRANSACTION_HISTORY_TO_FILE.md

```diff
@@ -0,0 +1,114 @@
+# Iteration 1 (basics)
+
+## Abstract
+
+Users want to export their transaction history into an Excel document.
+
+Default format gotta be XLSX, but optionally we might also support CSV, cuz it's more universal and might be imported into Excel as well as lots of other programs.
+
+## UI
+
+I suggest adding a new "Export to File" icon at the top-right of the transactions list. Something like this (general idea):
+
+![image](https://user-images.githubusercontent.com/5585355/50343034-8e94cc80-0536-11e9-9058-1af18be67f8a.png)
+
+When user clicks "Export to file" - new dialog window opens up:
+
+![image](https://user-images.githubusercontent.com/5585355/50401206-c6ea1400-079d-11e9-9980-2fb1c9cbc493.png)
+
+
+Dialog contains:
+1. Comment about the fact that the whole existing history will be exported
+2. Dropdown-selector: "File format"
```

Comment:
I can assume max at we will support following file-format:
1. xlsx
2. csv
3. pdf
4. json ( but can't see any use case for json)

If total max possible supported file format is only 3 or 4 then It can be easily fit into one line.
changing that to one line will have following benefits:
1. Format selection will be max 0 or 1 click operation.
2. Every option will be visible to the used at it will be faster to select.

Reply:
I did not manage yet, but plan to do it today - we will add another iteration to the spec that introduces more various formats to the export. Different formats in this case will not mean different **file** types, but rather different vendor formats (for compatibility with different websites that allow transaction history tracking). E.g. "Site1-CSV", "Site1-XLSX", "Site2-CSV", "Site2-XLSX", etc.

So it's preferable to make design in such a way that new formats might be easily added without changing anything visually.